mne/decoding/csp.py

```diff
@@ -86,22 +91,14 @@ class CSP(TransformerMixin, BaseEstimator):
     """
 
     def __init__(self, n_components=4, reg=None, log=None, cov_est="concat",
-                 transform_into='average_power', norm_trace=False):
+                 transform_into='average_power', norm_trace=False,
+                 method_params=None):
```

Comment:
method_params in a CSP class does not suggest it's related to cov estimation.

I would stick to the reg callable and expert users can do a functools.partial to set their custom params.

Reply:
It makes these methods less like the `compute_covariance` equivalent (why have `method_params` passed separately there but not here?) but I'll change it.

Reply:
Thinking about it more, removing `method_params` and forcing people to use `partial` will have a couple of negative consequences:

1. The `cov.py` code will need to be changed to handle `callable`, which will (based on its current design) make it more complex / harder to understand.
2. This change will make the `compute_covariance` and `decoding` methods less similar -- a big goal was to unify them as much as possible.

A simpler alternative that avoids these drawbacks is change `method_params` to match the `reg` name -> `reg_params`. Then in `compute_covariance` we have `method` and `method_params`, in decoding we have `reg` and `reg_params`, which seems straightforward enough to me.

Reply:
I would call it:

cov_method_params